logstash-core/lib/logstash/plugin.rb

```diff
@@ -92,7 +92,7 @@ def debug_info
   end
 
   def metric=(new_metric)
-    @metric = new_metric.namespace(@id)
+    @metric = new_metric
```

Comment:
This is interesting change, you moved the responsibility to namespace the metric with plugin id from the same plugin to the user of this plugin interface, isn't ? any reason for this? I see this as a more natural behaviour of the plugin base class as we aim to have all plugins request namespaced.